### ISSUE TITLE
Scheduler called at each step, instead of epoch.

### ISSUE THREAD

**Issue by vincentqb** [pytorch/vision#1261]
I noticed in the [tutorial](https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html) that the scheduler is called at the end of each epoch, as expected. However, I noticed that `train_one_epoch` also calls the scheduler after each image: see [here](https://github.com/pytorch/vision/blob/master/references/detection/engine.py#L49). Do I understand engine correctly?

**Comment by prajjwal1** [comment]
Yes, it's a batch of image tensors (not just one image if bs>1). `lr_scheduler` is being called twice.

**Comment by EmilioOldenziel** [comment]
Hi @vincentqb and @prajjwal1 .
See issue #1213 for the answer to your question.

**Comment by fmassa** [comment]
Exactly as @EmilioOldenziel mentioned, thanks for the reply!
